File: webpack_src/data/Good.ts
export type Goods = {
    _embedded: {
        groups: Good[]
    }
}

export type Good = {
    id?: number,
    name: String,
    deleted: boolean,
    subGroup?: string,
    _links: {
        self: { href: string },
        good: { href: string },
        subGroup?: { href: string }
    }
}

File: src/main/java/com/kehsa/spring_rest/database/entities/Good.java
package com.kehsa.spring_rest.database.entities;

import lombok.*;

import javax.persistence.*;

@Entity
@Table(name = "goods")
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
@EqualsAndHashCode
public class Good {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String name;
    private boolean deleted;
    @ManyToOne
    private SubGroup subGroup;
}